course.py

```diff
@@ -262,16 +266,38 @@ def add_duplicate_course():
     return redirect(url_for('course.course_info', course_id=course_id, year=year))
 
 
+@course_bp.route('/evaluation/<int:evaluation_id>')
+@login_required
+def course_evaluation(evaluation_id):
+    evaluation = db.session.query(Evaluation).get(evaluation_id)
+    courses = db.session.query(Course).filter_by(year=evaluation.course_year).all()
+
+    course = evaluation.course
+    user = db.session.query(User).filter_by(id=session['user_id']).first()
+    teachers = course.course_teacher
+
+    # Accessible only to the admin, the evaluation creator and the course teachers
+    if (not user.is_admin) and (user.id != evaluation.user_id) and (user.id not in [teacher.user_id for teacher in teachers]):
+        flash("You are not allowed to access this page", "danger")
+        return redirect(url_for('course.courses', course_id=course.id, year=course.year))
+
+    return render_template('evaluations.html', evaluation=evaluation, current_year=evaluation.course_year,
+                           courses=courses)
+
+
 @course_bp.route('/evaluations/<int:user_id>/<int:current_year>')
 @login_required
-def evaluations(user_id, current_year):
+@check_access_level(Role.RESEARCHER)
+def user_evaluation(user_id, current_year):
     courses = db.session.query(Course).filter_by(year=current_year).all()
 
-    return render_template('evaluations.html', courses=courses, current_year=current_year, user_id=user_id)
+    return render_template('evaluations.html', courses=courses, current_year=current_year, user_id=user_id,
+                           evaluation=None)
 
 
 @course_bp.route('/create_evaluations/<int:user_id>/<int:current_year>', methods=['POST'])
```

Comment:
There is a security issue here: we can create an evaluation for someone else by editing the form `action` link. User id should be fetched from session information.